uaclient/entitlements/fips.py

```diff
@@ -12,6 +12,74 @@
 event = event_logger.get_event_logger()
 
 
+FIPS_CONDITIONAL_PACKAGES = {
+    "xenial": [
+        "strongswan",
+        "strongswan-hmac",
+        "openssh-client",
+        "openssh-client-hmac",
+        "openssh-server",
+        "openssh-server-hmac",
+    ],
+    "bionic": [
+        "strongswan",
+        "strongswan-hmac",
+        "openssh-client",
+        "openssh-client-hmac",
+        "openssh-server",
+        "openssh-server-hmac",
+    ],
+    "focal": [
+        "strongswan",
+        "strongswan-hmac",
+        "openssh-client",
+        "openssh-server",
+    ],
+}
+# In containers, we don't install the ubuntu-fips
+# metapackage, but we do want to auto-upgrade any
+# fips related packages that are already installed.
+# These lists need to be kept up to date with the
+# Depends of ubuntu-fips.
+FIPS_CONTAINER_CONDITIONAL_PACKAGES = {
+    "xenial": [
+        "strongswan",
+        "strongswan-hmac",
+        "openssh-client",
+        "openssh-client-hmac",
+        "openssh-server",
+        "openssh-server-hmac",
+        "openssl",
+        "libssl1.0.0",
+        "libssl1.0.0-hmac",
+    ],
+    "bionic": [
+        "strongswan",
+        "strongswan-hmac",
+        "openssh-client",
+        "openssh-client-hmac",
+        "openssh-server",
+        "openssh-server-hmac",
+        "openssl",
+        "libssl1.1",
+        "libssl1.1-hmac",
+        "libgcrypt20",
+        "libgcrypt20-hmac",
```

Comment:
For all of the extra packages here, I am assuming those are installed by the metapackage.
If that is true, then we would have a problem if the metapackage starts depending on another package, since we would require an SRU to update this list.

We could parse the dependencies of the `ubuntu-fips` package, but it has other dependencies there that we would need to remove, like `grub2-common` for example.

This would guarantee that we would have the updated packages whenever we need them. But it may be an overkill, since I don't expect the metadata dependencies to change. But let me know if you think this is worth it

Reply:
Right, these additional packages are from the metapackage.

I'm not sure if parsing the metapackage deps is worth it. I don't think a delay in updating this list via SRU is a major problem because we should recommend anyone who enabled fips to run `apt upgrade` anyway, and that should capture any fips upgrades that are necessary.

But I'm open to other opinions - what do you think?

Reply:
Oh yes we should, I'll add it